packages/api/tests/lib/rules/test-updateRuleResources.js

```diff
@@ -0,0 +1,475 @@
+  t.false(createdRule.arn === undefined);
+  t.false(createdRule.log_event_arn === undefined);
+
+  // update rule state by setting enabled to false
+  const updatedKinesisRule = {
+    ...createdRule,
+    enabled: false,
+  };
+  const updatedRule = await updateRuleTrigger(kinesisRule, updatedKinesisRule, testKnex);
+  t.false(updatedRule.enabled);
+
+  const updatedKinesisEventMappings = await getKinesisEventMappings();
+  const updatedconsumerEventMappings = updatedKinesisEventMappings[0].EventSourceMappings;
+  const updatedlogEventMappings = updatedKinesisEventMappings[1].EventSourceMappings;
+
+  t.is(updatedconsumerEventMappings.length, 1);
+  t.is(updatedlogEventMappings.length, 1);
+  t.is(updatedconsumerEventMappings[0].UUID, updatedRule.arn);
+  t.is(updatedlogEventMappings[0].UUID, updatedRule.log_event_arn);
+
+  t.is(createdRule.arn, updatedRule.arn);
+  t.is(createdRule.log_event_arn, updatedRule.log_event_arn);
+
+  // Clean Up
+  t.teardown(async () => {
+    await deleteRuleResources(testKnex, createdRule);
+    await deleteRuleResources(testKnex, updatedRule);
+  });
+});
+
+test.serial('Updating a kinesis type rule workflow does not affect value or event source mappings', async (t) => {
+  const { testKnex } = t.context;
+  const kinesisRule = fakeRuleRecordFactory({
+    type: 'kinesis',
+    workflow,
+    enabled: true,
+    value: randomString(),
+  });
+  // create rule trigger
+  const createdRule = await createRuleTrigger(kinesisRule);
+  const kinesisEventMappings = await getKinesisEventMappings();
+  const consumerEventMappings = kinesisEventMappings[0].EventSourceMappings;
+  const logEventMappings = kinesisEventMappings[1].EventSourceMappings;
+
+  t.is(consumerEventMappings.length, 1);
+  t.is(logEventMappings.length, 1);
+  t.is(consumerEventMappings[0].UUID, createdRule.arn);
+  t.is(logEventMappings[0].UUID, createdRule.log_event_arn);
+
+  // update rule workflow
+  const updatedKinesisRule = {
+    ...createdRule,
+    workflow: 'new-workflow',
+  };
+  const updatedRule = await updateRuleTrigger(kinesisRule, updatedKinesisRule, testKnex);
+  const updatedKinesisEventMappings = await getKinesisEventMappings();
+  const updatedconsumerEventMappings = updatedKinesisEventMappings[0].EventSourceMappings;
+  const updatedlogEventMappings = updatedKinesisEventMappings[1].EventSourceMappings;
+
+  t.is(updatedconsumerEventMappings.length, 1);
+  t.is(updatedlogEventMappings.length, 1);
+  t.is(updatedconsumerEventMappings[0].UUID, updatedRule.arn);
+  t.is(updatedlogEventMappings[0].UUID, updatedRule.log_event_arn);
+
+  t.is(createdRule.arn, updatedRule.arn);
+  t.is(createdRule.log_event_arn, updatedRule.log_event_arn);
+
+  // Clean Up
+  t.teardown(async () => {
+    await deleteRuleResources(testKnex, createdRule);
+    await deleteRuleResources(testKnex, updatedRule);
+  });
+});
+
+test.serial('Updating a valid SQS rule to have an invalid schema throws an error and does not update triggers', async (t) => {
```

Comment:
```suggestion
test.serial('Updating a valid SQS rule to have an invalid schema throws an error', async (t) => {
```

There are no triggers for SQS rules that get updated on the rule (no `arn` property is set)

Reply:
[`aa59c90` (#2871)](https://github.com/nasa/cumulus/pull/2871/commits/aa59c90a42ee176d4075b4de2c1bf78678177b09#diff-057c67806291c57c6410d819b57bda8d87b5370152a299e6baca44b5a44f0e40L299)